programs/move_loader_api/src/lib.rs

```diff
@@ -79,13 +99,31 @@ pub fn execute(
     let main_module = script.into_module();
     let module_id = main_module.self_id();
     module_cache.cache_module(main_module);
-    let txn_metadata = TransactionMetadata::default();
+    let mut txn_metadata = TransactionMetadata::default();
+    txn_metadata.sender = invoke_info.sender_address;
 
     let mut vm = TransactionExecutor::new(&module_cache, data_store, txn_metadata);
-    let result = vm.execute_function(&module_id, &"main".to_string(), arguments_to_locals(args));
+    let result = vm.execute_function(
+        &module_id,
+        &"main".to_string(),
```

Comment:
Awesome they factored this out. We could do the same. Currently we have `LoaderInstruction::InvokeMain(args)`. We could have `LoaderInstruction::Invoke(name, args)`.